docs/usage/shoot_high_availability_best_practices.md

```diff
@@ -0,0 +1,431 @@
+# Implementing High Availability and Tolerating Zone Outages
+
+Implementing highly available software that can tolerate even a zone outage unscathed is no trivial task. You will find here various recommendations to get closer to that goal. While many recommendations are general enough, the examples are specific in how to achieve this in a Gardener-managed cluster and where/how to tweak the different control plane components. If you do not use Gardener, it may be still a worthwhile read.
+
+First however, what is a zone outage? It sounds like a clear-cut "thing", but it isn't. There are many things that can go haywire. Here are some examples:
+
+- Elevated cloud provider API error rates for individual or multiple services
+- Network bandwidth reduced or latency increased, usually also effecting storage sub systems as they are network attached
+- No networking at all, no DNS, machines shutting down or restarting, ...
+- Functional issues, of either the entire service (e.g. all block device operations) or only parts of it (e.g. LB listener registration)
+- All services down, temporarily or permanently (the proverbial burning down data center :fire:)
+
+This and everything in between make it hard to prepare for such events, but you can still do a lot. The most important recommendation is to not target specific issues too much - tomorrow another service will fail in an unanticipated way. Rather focus on [meaningful availability](https://research.google/pubs/pub50828) than on internal signals (useful, but not as relevant as the former). Always prefer automation over manual intervention (e.g. leader election is a pretty robust mechanism, auto-scaling may be required as well, etc.).
+
+Also remember that HA is costly - you need to balance it against the cost of an outage as silly as this may sound, e.g. running all this excess capacity "just in case" vs. "going down" vs. a risk-based approach in between where you have means that will kick in, but they are not guaranteed to work (e.g. if the cloud provider is out of resource capacity). Maybe some of your components must run at the highest possible availability level, but others not - that's a decision only you can make.
+
+## Control Plane
+
+The Kubernetes cluster control plane is managed by Gardener (as pods in separate infrastructure clusters to which you have no direct access) and can be set up with no failure tolerance (control plane pods will be recreated best-effort when resources are available) or one of the [failure tolerance types `node` or `zone`](https://github.com/gardener/gardener/blob/master/docs/usage/shoot_high_availability.md).
```

Comment:
Nit: You could use relative link. Just to make sure that when someone is browsing the docs at tag vX.Y.Z will be redirected to the same version of the referenced document (and not master).

```suggestion
The Kubernetes cluster control plane is managed by Gardener (as pods in separate infrastructure clusters to which you have no direct access) and can be set up with no failure tolerance (control plane pods will be recreated best-effort when resources are available) or one of the [failure tolerance types `node` or `zone`](shoot_high_availability.md).
```

Reply:
@ialidzhikov I will file another commit and "fix" that.

For the record though, it was deliberate and I weighted the pros and cons and decided against it, but if that's how it should be done in this repo, I will of course comply.

The reasons I find it "problematic", although your points are absolutely valid and it's how "it's done":
- The document is then bound to this repo and can be read only "in GitHub"
- This document specifically is one that people may want to "rip out" and save as MD file, which will then not work and all Gardener links will be dead
- Consequently, if I open the file locally, the links are dead and the error message is what is expected, e.g.: "File not found at path:
/docs/usage/shoot_high_availability.md"
- Link checkers must be aware or else they will fail

As I plan to post this document as blog, I will now replace the absolute with relative links and then replace the relative links again with absolute links. No big deal, `sed` is my friend, but I am still not convinced this is overall the right approach.

Having said that, I am using relative links myself, e.g. in the repo where this document was originally written (https://github.com/gardener/chaos-engineering), but not for this document. Maybe, inconsistent, but not everything is black and white.

However, following a repo guideline is important/generates value. If this repo (and all Gardener repos) require relative links, relative links it will be. :-)

@rfranzke mentioned something odd though, which is that absolute links won't work with `docforge`, which is why follow-up commits like https://github.com/gardener/gardener/pull/7394/files#diff-b06e6db607964c79c376846a413d64583475deff87d705aacac96f7ca26975afR67 are done by @n-boshnakov. Is that so? I wasn't aware/wouldn't have expected that.

Reply:
Yes, I agree that we don't have documented conventions for this.  If there is already one, @n-boshnakov @Kostov6 @dimitar-kostadinov feel free to link it. `docforge` should handle all links - relative and absolute.
I think the problem is that we don't have conventions similar to https://kubernetes.io/docs/contribute/style/style-guide/. I myself raised similar question for missing conventions for which one should be used in the docs - "Shoot" or "shoot". So, I rather think that a convention for this question should be added.

Reply:
The unwritten guidelines we follow for Gardener documentation in regards to linking Github resources is "resource is in the same repo -> relative link", "resource is in another repo -> absolute link". But you are right, there is no documented convention for using absolute vs relative links. I will update the [Formatting Guide](https://github.com/gardener/documentation/blob/master/website/documentation/contribute/20_documentation/20_formatting_guide/_index.md) to include a note on absolute vs relative link usage.